docs/sources/flow/reference/components/discovery.file.md

```diff
@@ -0,0 +1,76 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.file
+title: discovery.file
+---
+
+# discovery.file
+
+`discovery.file` discovers files on the local filesystem using the [doublestar][] library.
+
+[doublestar]: https://github.com/bmatcuk/doublestar
+
+## Usage
+
+```river
+discovery.file "LABEL" {
+  paths = ["DOUBLESTAR_PATH"]
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type           | Description                                                                                                      | Default | Required
+---- |----------------|------------------------------------------------------------------------------------------------------------------|-----| --------
+`paths` | `list(map(string))` | Doublestar-compatible paths to search, looks for keys with `__path__`.                                           |     | yes
```

Comment:
Should we expand on this a little more, maybe in the intro paragraph? We can talk about how the component receives a set of targets which _must_ define the `__path__` key, similar to how Prometheus uses `__address__`, but can have other identifying labels accompanying it etc.